src/composer.ts

```diff
@@ -820,28 +1132,42 @@ export class Composer<C extends Context> implements MiddlewareObj<C> {
 }
 
 // === Util functions and types
-function triggerFn(trigger: MaybeArray<string | RegExp>) {
-    return toArray(trigger).map((t) =>
-        typeof t === "string"
-            ? (txt: string) => (txt === t ? t : null)
-            : (txt: string) => t.exec(txt)
-    );
+type StringExecFn = (content: string) => string | null;
+type RegExpExecFn = (content: string) => RegExpExecArray | null;
+
+function isNonEmptyStringArray(xs: any): xs is NonEmptyArray<string> {
+    return Array.isArray(xs) && xs.length > 0 &&
+        xs.every((x) => typeof x === "string");
+}
+function isNonEmptyRegExpArray(xs: any): xs is NonEmptyArray<RegExp> {
+    return Array.isArray(xs) && xs.length > 0 &&
+        xs.every((x) => x instanceof RegExp);
+}
```

Comment:
Why is this distinction even necessary? It worked well before, it is just for type annotations?

Reply:
Yup. It's no longer required, and deleted.